pkg/component/etcd/poddisruptionbudget/poddisruptionbudget.go

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package poddisruptionbudget
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Masterminds/semver"
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/etcd-druid/pkg/utils"
+
+	"github.com/gardener/gardener/pkg/controllerutils"
+	gardenercomponent "github.com/gardener/gardener/pkg/operation/botanist/component"
+	policyv1 "k8s.io/api/policy/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type component struct {
+	client               client.Client
+	namespace            string
+	values               *Values
+	controllerConfig     *rest.Config
+	k8sVersionGetterFunc K8sVersionGetter
+}
+
+// K8sVersionGetter defines a func type that accepts a rest configuration and returns the semantic k8s version of the k8s cluster
+type K8sVersionGetter func(config *rest.Config) (*semver.Version, error)
```

Comment:
Instead of using this function and increasing the complexity, Can we just pass it to `New` and store it in the `Values` of the component? This is what we do for every component in gardener as well.

Reply:
Updated